modules/dnn/src/layers/layers_common.cpp

```diff
@@ -144,33 +144,86 @@ void getStrideAndPadding(const LayerParams &params, std::vector<size_t>& pads_be
 }
 }
 
-void getPoolingKernelParams(const LayerParams &params, std::vector<size_t>& kernel, bool &globalPooling,
+void getPoolingKernelParams(const LayerParams &params, std::vector<size_t>& kernel, std::vector<bool>& globalPooling,
                             std::vector<size_t>& pads_begin, std::vector<size_t>& pads_end,
                             std::vector<size_t>& strides, cv::String &padMode)
 {
-    globalPooling = params.has("global_pooling") &&
-                    params.get<bool>("global_pooling");
+    bool is_global = params.get<bool>("global_pooling", false);
+    globalPooling.resize(3);
+    globalPooling[0] = params.get<bool>("global_pooling_d", is_global);
+    globalPooling[1] = params.get<bool>("global_pooling_h", is_global);
+    globalPooling[2] = params.get<bool>("global_pooling_w", is_global);
 
-    if (globalPooling)
+    is_global = globalPooling[0] || globalPooling[1] || globalPooling[2];
+    if (!is_global)
     {
-        util::getStrideAndPadding(params, pads_begin, pads_end, strides, padMode);
-        if(params.has("kernel_h") || params.has("kernel_w") || params.has("kernel_size"))
-        {
+        util::getKernelSize(params, kernel);
+        util::getStrideAndPadding(params, pads_begin, pads_end, strides, padMode, kernel.size());
+    }
+    else
+    {
+        if ((globalPooling[0] && params.has("kernel_d")) ||
+            (globalPooling[1] && params.has("kernel_h")) ||
+            (globalPooling[2] && params.has("kernel_w")) ||
+            params.has("kernel_size")) {
             CV_Error(cv::Error::StsBadArg, "In global_pooling mode, kernel_size (or kernel_h and kernel_w) cannot be specified");
         }
-        for (int i = 0; i < pads_begin.size(); i++) {
-            if (pads_begin[i] != 0 || pads_end[i] != 0)
-                CV_Error(cv::Error::StsBadArg, "In global_pooling mode, pads must be = 0");
+
+        kernel.resize(3);
+        pads_begin.resize(3, 0);
+        pads_end.resize(3, 0);
+        strides.resize(3, 1);
+        kernel[0] = params.get<int>("kernel_d", 1);
+        kernel[1] = params.get<int>("kernel_h", 1);
+        kernel[2] = params.get<int>("kernel_w", 1);
+
+        pads_begin[1] = params.get<int>("pad_t", 0);
+        pads_begin[2] = params.get<int>("pad_l", 0);
+        pads_end[1] = params.get<int>("pad_b", 0);
+        pads_end[2] = params.get<int>("pad_r", 0);
```

Comment:
Let's try once again to simplify it considering the following rules:
1) Paddings. User should specify either `pad` or `pad_w` && `pad_h` (getParameter) or `pad_l` && `pad_t` && `pad_r` && `pad_b` (current 3.4 works in this way.). So in case of global pooling by a specific axis, all the other padding will be specified anyway.
2) Strides similar as (1) but with `stride` or `stride_w` && `stride_h`
3) check only parameters which conflict with each other:
```
(!params.has("global_pooling") || !params.has("kernel_size")) &&
(!params.has("global_pooling_d") || !params.has("kernel_size_d")) &&
(!params.has("global_pooling_h") || !params.has("kernel_size_h")) &&
(!params.has("global_pooling_w") || !params.has("kernel_size_w"))
```